tests/validation/tests/v3_api/Jenkinsfile_deploy_and_test

```diff
@@ -82,11 +86,29 @@ node {
               "&& export RANCHER_CLUSTER_NAME=${env.CLUSTER_NAME} " + "&& pytest -v -s --junit-xml=${testResultsOut} " +
               "${PYTEST_OPTIONS} ${testsDir}\'"
           } catch(err) {
+            RANCHER_DEPLOYED = false
             echo "Error: " + err
             echo 'Test run had failures. Collecting results...'
+          }finally {
+        stage('Delete Rancher Server') {
+          try {
+             if (env.RANCHER_DELETE_SERVER.toLowerCase() == "true" && RANCHER_DEPLOYED) {
+                sh "docker run --name ${deleteContainer} -t --env-file ${envFile} " +
+                "${imageName} /bin/bash -c \'export CATTLE_TEST_URL=${env.CATTLE_TEST_URL} && " +
+                "export ADMIN_TOKEN=${env.ADMIN_TOKEN} && export USER_TOKEN=${env.USER_TOKEN} &&"+
+                 "pytest -v -s --junit-xml=${deleteResultsOut} " +
+                "${deletePytestOptions} ${testsDir}\'"
+                }
+            else {
+              echo "Rancher server not deployed, skipping delete."
+            }
+          } catch(err) {
+            echo "Error: " + err
+            currentBuild.result = 'FAILURE'
           }
         }
-
+      }
+      }
         stage('Test Report') {
           // copy and archive test results
           sh "docker cp ${setupContainer}:${rootPath}${setupResultsOut} ."
```

Comment:
Also copy the reports from the delete container if the server was deployed

Reply:
Added report for delete container